src/main/java/co/rsk/bitcoinj/script/Script.java

```diff
@@ -197,10 +197,10 @@ public List<ScriptChunk> getChunks() {
      * Bitcoin Core does something similar.</p>
      */
     private void parse(byte[] program) throws ScriptException {
-        ScriptParserResult result = ScriptParser.parseScriptProgram(program);
-        chunks = result.getChunks();
-        if (result.getException().isPresent()) {
-            throw new ScriptException("There was an error parsing the script", result.getException().get());
+        try {
+            chunks = ScriptParser.parseScriptProgram(program);
+        } catch (Exception ex) {
```

Comment:
Is this right? What exceptions does `parseScriptProgram` throw?

Reply:
It throws `ScriptException` when the program does not meet the expected format. It is the same logic as in [bitcoinj](https://github.com/rsksmart/bitcoinj/blob/3e0a65f36f10062b8b3247204715b6b370158666/core/src/main/java/org/bitcoinj/script/Script.java#L194)